### PR TITLE
Playlist feature done

### PR DESCRIPTION
**Pull request by animeshgarai09**
This PR adds playlists feature for my video library.
So check only playlist prefixed files.

Thank you

### CODE REVIEW

File: src/hooks/useShareVideo.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        if (navigator && navigator.clipboard && navigator.clipboard.writeText) {
+            toast({
+                status: "success",
+                message: "Link copied to clipboard",
+                duration: 2
+            })
+            return navigator.clipboard.writeText(url);
+        }
+        return Promise.reject('The Clipboard API is not available.');
+    }
```
Comment (ShekharDhangar) [inline src/hooks/useShareVideo.js:16]: Appreciation: Implementation of copy link feature. Its really nice

**Review by ShekharDhangar — APPROVED** [review]
Really good project, very clean UI and nice implementation of copy link. Overall really nice.

File: src/store/providers/playlist-provider.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+                type: "SET_PLAYLISTS",
+                payload: response.data.playlists
+            })
+            if (PlaylistState.selectedPlaylistId === null) {
+                PlaylistDispatch({
+                    type: "SET_SELECTED_PLAYLIST",
+                    payload: response.data.playlists[0]._id
+                })
+            }
+
```
Comment (vrinda-mahajan) [inline src/store/providers/playlist-provider.js:80]: Suggestion:- You can use `async-await` and `try-catch` block for API calls.

File: src/pages/playlists/playlists.module.scss [hunk 3]
```diff
@@ -61,5 +67,29 @@
...
+    transition: 0.2s;
+    input {
+        margin-right: 1rem;
+    }
+    &:hover {
+        color: var(--clr-gray-3);
     }
+    // &:last-child {
+    //     margin-bottom: 1rem;
+    // }
```
Comment (vrinda-mahajan) [inline src/pages/playlists/playlists.module.scss:94]: Remove the commented code, if not required.

**Review by vrinda-mahajan — APPROVED** [review]
Nothing much to suggest. The code looks clean and UI is amazing💯. Good to merge.

**Pull request merged**